Repository/Courses/CourseRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Interface.Repositories;
+using Model.DTOs;
+using Repository.DbModels;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Repository.Courses
+{
+    public class CourseRepository : ICourseRepository
+    {
+        private readonly DbModels.DbModels context;
+
+        public CourseRepository(DbModels.DbModels context)
+        {
+            this.context = context;
+        }
+
+        public CourseDTO Add(CourseDTO course) 
+        {
+            context.Courses.Add(new DbModels.Course
+            {
+                 ID = course.ID,
+                 Credits = course.Credits,
+                 Title = course.Title,
+            });
+            context.SaveChanges();
+            return course;
+        }
+
+        public CourseDTO Delete(int ID)
+        {
+            Course course = context.Courses.Find(ID);
+            if (course != null)
+            {
+                var deletetCourse = new CourseDTO 
+                {
+                    ID = course.ID,
+                    Credits = course.Credits,
+                    Title = course.Title,
+                };
+                context.Courses.Remove(course);
+                context.SaveChanges();
+                return deletetCourse;
+            }
+            return null;
+        }
+
+        public CourseDTO GetCourseByID(int ID)
+        {
+            var existingCourse = context.Courses.Find(ID);
+            if (existingCourse != null)
+            {
+                return new CourseDTO
+                {
+                    ID = existingCourse.ID,
+                    Credits = existingCourse.Credits,
+                    Title = existingCourse.Title,
+                };
+            }
+            return null;
+        }
+
+        public IEnumerable<CourseDTO> GetCourses()
+        {
+            var courses = context.Courses;
```

Comment:
This could be simplified - `return context.Courses.Select(c => new CourseDTO
                {
                    ID = c.ID,
                    Credits = c.Credits,
                    Title = c.Title,
                }))`